pkg/snapshot/snapshot_azure.go

```diff
@@ -0,0 +1,172 @@
+package snapshot
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+)
+
+type azureSnapshotter struct {
+	config    *AzureConfig
+	container azblob.ContainerClient
+}
+
+// AzureConfig contains the configuration options for storing database
+// snapshots in Azure Storage accounts.
+type AzureConfig struct {
+	// AccountName is the Azure account name.
+	AccountName string
+
+	// AccountKey is the secret to access the storage account.
+	AccountKey string
+
+	// StorageAccount is the storage account name.
+	StorageAccount string
+
+	// ContainerName is the top level namespace where we'll keep snapshots.
+	ContainerName string
+
+	// Timeout controls how long we wait for Snapshotter.Save or Snapshotter.Load
+	// to finish.
+	Timeout time.Duration
+
+	// Maximum number of times to retry an upload or download.
+	Retries int
+}
+
+// NewAzureSnapshotter takes a pointer to AzureConfig and returns a type that
+// satifies the Snapshotter interface.
+func NewAzureSnapshotter(config *AzureConfig) (Snapshotter, error) {
+	cred, err := azblob.NewSharedKeyCredential(config.AccountName, config.AccountKey)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("https://%s.blob.core.windows.net/", config.StorageAccount)
```

Comment:
Ideally we wouldn't build this URL since some users will be using Azure Government, which will use URLs like `blob.core.usgovcloudapi.net`

Reply:
This is now set in SnapshotBackupURL. The URL still gets "built" but the hostname is fully customizable.